lib/ansible/modules/system/java_keystore.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Guillaume Grossetie <ggrossetie@yuzutech.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: java_keystore
+short_description: Create or delete a Java keystore in JKS format
+description:
+     - Create or delete a Java keystore in JKS format for a given certificate
+version_added: "2.5"
+options:
+    name:
+        description:
+          - Name of the certificate
+        required: true
+    certificate:
+        description:
+          - Certificate that should be used to create the key store
+        required: false
+    private_key:
+        description:
+          - Private key that should be used to create the key store
+        required: false
+    password:
+        description:
+          - Password that should be used to secure the key store
+        required: false
+    dest:
+        description:
+          - Absolute path where the jks should be generated.
+        required: true
+    owner:
+        description:
+          - Name of the user that should own jks file
+        required: false
+    group:
+        description:
+          - Name of the group that should own jks file
+        required: false
+    mode:
+        description:
+          - Mode the file should be
+        required: false
+    force:
+        description:
+          - Key store will be created even if it already exists
+        required: false
+requirements: [openssl, keytool]
+author: Guillaume Grossetie
+'''
+
+EXAMPLES = '''
+# Create a key store for the given certificate
+java_keystore:
+  certificate: |
+    -----BEGIN CERTIFICATE-----
+    h19dUZ2co2fI/ibYiwxWk4aeNE6KWvCaTQOMQ8t6Uo2XKhpL/xnjoAgh1uCQN/69
+    MG+34+RhUWzCfdZH7T8/qDxJw2kEPKluaYh7KnMsba+5jHjmtzix5QIDAQABo4IB
+    -----END CERTIFICATE-----
+  private_key: |
+    -----BEGIN RSA PRIVATE KEY-----
+    DBVFTEVDVFJJQ0lURSBERSBGUkFOQ0UxFzAVBgNVBAsMDjAwMDIgNTUyMDgxMzE3
+    GLlDNMw/uHyME7gHFsqJA7O11VY6O5WQ4IDP3m/s5ZV6s+Nn6Lerz17VZ99
+    -----END RSA PRIVATE KEY-----
+  password: changeit
+  dest: /etc/security/keystore.jks
```

Comment:
I think it would also be nice to show an example using a lookup plugin to read the keys from a file so that these don't have to be inlined.

```
java_keystore:
  certificate: "{{lookup('file', '/path/to/certificate.crt') }}"
  private_key: "{{lookup('file', '/path/to/private.key') }}"
  password: changeit
  dest: /etc/security/keystore.jks
```